lib/trento/application/integration/discovery/payloads/cluster/cib_discovery_payload.ex

```diff
@@ -0,0 +1,171 @@
+defmodule Trento.Integration.Discovery.ClusterDiscoveryPayload.Cib do
+  @moduledoc """
+  Cib field payload
+  """
+  alias Trento.Support.ListHelper
+
+  defmodule Primitive do
+    @moduledoc """
+    Primitive field payload
+    """
+
+    @required_fields [:id, :type, :class, :operations, :instance_attributes]
+    use Trento.Type
+
+    deftype do
+      field :id, :string
+      field :type, :string
+      field :class, :string
+      field :provider, :string
+
+      embeds_many :operations, Operation, primary_key: false do
+        field :id, :string
+        field :name, :string
+        field :role, :string
+        field :timeout, :string
+        field :interval, :string
+      end
+
+      embeds_many :instance_attributes, InstanceAttribute, primary_key: false do
+        field :id, :string
+        field :name, :string
+        field :value, :string
+      end
+    end
+
+    defp transform_nil_lists(
```

Comment:
I don't know if it can happen, but maybe we should add a safe `transform_nil_lists` implementation for the cases where we receive something unknown.
Something like:
```
defp transform_nil_lists(attrs), do: attrs
```

What do you think?
```